File: src/main/java/java_examples/1_declarative_vs_imperative.java
package java_examples;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

class Declarative_vs_imperative {

    public static void main(String[] args) {
        List<String> list_of_strings = Arrays.asList(
                "small",
                "alsosmall",
                "quiteabitbigger",
                "definitelytoolargetokeepattentionofreader"
        );

        List<String> filtered_list = new ArrayList<>();
        for (String item : list_of_strings) {
            if (item.length() < 10) {
                filtered_list.add(item);
            }
        }

        System.out.println(filtered_list);
    }

}

File: python/1_declarative_vs_imperative.py
list_of_strings = [
    "small",
    "alsosmall",
    "quiteabitbigger",
    "definitelytoolargetokeepattentionofreader"
]

filtered_list = list()
for item in list_of_strings:
    if len(item) < 10:
        filtered_list.append(item)

print(filtered_list)
